discord_connector.py

```diff
@@ -339,6 +339,46 @@ def parse_message_flags(self, message):
 
         return true_flags
 
+    def _handle_delete_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        message, ret_val = self._async_loop.run_until_complete(self.delete_message(channel_id, message_id))
+        self.save_progress("Action on message {} ended with return value of: {}: {}".format(message_id, ret_val, message))
+
+        summary = action_result.update_summary({})
+        summary['action result: '] = "Action on message {} ended with return value of: {}: {}".format(message_id, ret_val, message)
+
+        if ret_val != 200:
+            return action_result.set_status(phantom.APP_ERROR)
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    async def delete_message(self, channel_id, message_id):
+        await self._client.login(self._token)
+
+        guild = await self._client.fetch_guild(self._guild_id)
```

Comment:
I start to wonder, if guild cannot be fetched once in `initialize()` because I see it being used in every single action anyway.

Reply:
it has been changed on kczernik_develop (the branch for PAPP-34496) to be a independent method called in initialize()